peps/pep-0751.rst

```diff
@@ -0,0 +1,901 @@
+  package, regardless of version
+
+
+``package.files.simple-repo-package-url``
+'''''''''''''''''''''''''''''''''''''''''
+
+- Optional (although mutually exclusive with
+  ``package.simple-repo-package-url``)
+- String
+- The value has the same meaning as ``package.simple-repo-package-url``
+- This key is avaible per-file to support :pep:`708` when some files override
+  what's provided by another `Simple Repository API`_ index
+
+
+``package.files.origin``
+''''''''''''''''''''''''
+
+- Optional
+- String
+- URI where the file was found when the lock file was generated
+- Useful for documenting where the file came from and potentially where to look
+  for the file if not already downloaded/available
+
+
+``package.files.hash``
+''''''''''''''''''''''
+
+- String
+- The hash value of the file contents using the hash algorithm specified by
+  ``hash-algorithm``
+- Used by installers to verify the file contents match what the locker worked
+  with
+
+
+``[package.vcs]``
+'''''''''''''''''
+
+- Must be specified if ``[[package.files]]`` is not (altough may be specified
+  simultaneously with ``[[package.files]]``)
+- Table representing the version control system containing the package and
+  version
+
+
+``package.vcs.type``
+''''''''''''''''''''
+
+- String
+- The type of version control system used
+- The valid values are specified by the
+  `registered VCSs <https://packaging.python.org/en/latest/specifications/direct-url-data-structure/#registered-vcs>`__
+  of the direct URL data structure
+
+
+``package.vcs.origin``
+''''''''''''''''''''''
+
+- String
+- The URI of where the repository was located when the lock file was generated
+
+
+``package.vcs.commit``
+''''''''''''''''''''''
+
+- String
+- The commit ID for the repository which represents the package and version
+- The value MUST be immutable for the VCS for security purposes
+  (e.g. no Git tags)
+
+
+``package.vcs.lock``
+'''''''''''''''''''''''''
+
+- Required when ``[[file-lock]]`` is used
+- An array of strings
+- An array of ``file-lock.name`` values which signify that the repository at the
+  specified commit is to be installed when the corresponding ``[[file-lock]]``
+  table applies to the environment
+- A name in the array may only appear if no file listed in
+  ``package.files.lock`` contains the name for the same package, regardless of
+  version
+
+
+``package.directory``
+'''''''''''''''''''''
+
+- Optional and only valid when ``[package-lock]`` is specified
+- String
+- A local directory where a source tree for the package and version exists
+- Not valid under ``[[file-lock]]`` as this PEP does not make an attempt to
+  specify a mechanism for verifying file contents have not changed since locking
+  was performed
+
+
+``[[package.build-requires]]``
+''''''''''''''''''''''''''''''
+
+- Optional
+- An array of tables whose structure matches that of ``[[package]]``
+- Each entry represents a package and version to use when building the
```

Comment:
Should this itself be a locked array? That is, if I add `a` here, and `a` depends on `b`, will `b` also be installed?

Reply:
Yes, and I thought that was implied by needing to match `[[package]]`. I'll make it more clear.